pilot/pkg/bootstrap/server.go

```diff
@@ -1198,3 +1203,43 @@ func (s *Server) initMeshHandlers() {
 		})
 	})
 }
+
+func (s *Server) initWorkloadTrustBundle() {
+	s.workloadTrustBundle = NewTrustBundle()
+	s.addStartFunc(func(stop <-chan struct{}) error {
+		go s.workloadTrustBundle.processUpdates(stop)
+		return nil
+	})
+
+	// MeshConfig: Add initial roots and callback
+	s.workloadTrustBundle.AddPeriodicMeshConfigUpdate(s.environment.Mesh())
+	s.environment.AddMeshHandler(func() {
+		s.workloadTrustBundle.AddPeriodicMeshConfigUpdate(s.environment.Mesh())
+	})
+
+	// IstioCA: Implicitly add roots corresponding to CA
+	if s.CA != nil {
+		rootCerts := []string{string(s.CA.GetCAKeyCertBundle().GetRootCertPem())}
+		s.workloadTrustBundle.EnqueueAnchorUpdate(&TrustAnchorUpdate{TrustAnchorConfig: TrustAnchorConfig{
+			source: SourceIstioCA,
+			certs:  rootCerts,
+		}})
+		s.addStartFunc(func(stop <-chan struct{}) error {
+			go s.workloadTrustBundle.AddPeriodicIstioCARootUpdate(stop, s.CA.GetCAKeyCertBundle(),
```

Comment:
2 comments here:
* 1hr seems like a *really* long time to have to wait for configuration to update
* I am not sure we need to poll here. Can we not have the code that is handling the updates just call some callback or similar?

Reply:
Makes sense. Given that the CA would immediately start signing certs with the new CA, the update to the TrustBundle has to happen as soon as root-certificate is rotated. In fact, the CA needs to receive an ack from all proxies before it can start using the new root of trust to sign certificates. Will add this to some callback.